src/client/Microsoft.Identity.Client/ManagedIdentity/ManagedIdentityClient.cs

```diff
@@ -81,5 +84,38 @@ internal static ManagedIdentitySource GetManagedIdentitySource()
                 return ManagedIdentitySource.DefaultToImds;
             }
         }
+
+        // Method to return true if a file exists and is not empty to validate the Azure arc environment.
+        private static bool ValidateAzureArcEnvironment(string identityEndpoint, string imdsEndpoint, ILoggerAdapter logger)
+        {
+            if (!string.IsNullOrEmpty(identityEndpoint) && !string.IsNullOrEmpty(imdsEndpoint))
+            {
+                logger?.Verbose(() => "[Managed Identity] Azure Arc managed identity is available through environment variables.");
+                return true;
+            }
+
+            string path;
+
+            if (DesktopOsHelper.IsWindows())
+            {
+                path = Environment.ExpandEnvironmentVariables("%Programfiles%\\AzureConnectedMachineAgent\\himds.exe");
+            }
+            else if (DesktopOsHelper.IsLinux())
+            {
+                path = "/opt/azcmagent/bin/himds";
+            }
+            else
+            {
+                return false;
```

Comment:
Consider adding a logging statement, maybe at warning level. As we don\t expect this logic to ever run elsewhere.